### PR TITLE
[Design] Support multiple SQL statements

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Design doc for https://github.com/sql-machine-learning/sqlflow/issues/695.

### CODE REVIEW

File: doc/support_multiple_sql_statements.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+### Splitting Location: Client-side vs Server-side
+
+While splitting at the client-side is relatively simple to implement. We prefer to split at the server-side for the following reasons.
+Loose coupling. SQLFlow server can use its lexer/parser to split SQL statements accurately. 
+Extensibility. SQLFlow server can run a sequence of SQL statements to form a workflow.
+
+### Splitting Technique: Hybrid Parser vs Lexer
+
+The hybrid parser solution uses the standard SQL parser and SQLFlow parser to determine the end of an SQL statement. The standard SQL parser first parses the extended SQL statement. It will raise error near SQLFlow extended keywords, like TRAIN and PREDICT. Then the SQLFlow parser starts from the error position and stops at the end of the first statement. However, this solution relies on the standard SQL parser to report the error **accurately** on the keywords, like TRAIN and PREDICT, that it can't recognize.
```
Comment (typhoonzero) [inline doc/support_multiple_sql_statements.md]: standard SQL parser => third-party parsers like tidb parser

File: doc/support_multiple_sql_statements.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Design Choice
+
+There are several design choices to make.
+
+### Splitting Location: Client-side vs Server-side
+
+While splitting at the client-side is relatively simple to implement. We prefer to split at the server-side for the following reasons.
+Loose coupling. SQLFlow server can use its lexer/parser to split SQL statements accurately. 
+Extensibility. SQLFlow server can run a sequence of SQL statements to form a workflow.
```
Comment (typhoonzero) [inline doc/support_multiple_sql_statements.md]: More details about what workflow is? We would like to analyze the dependencies of the execution order of all the SQL statements and form an execution graph automatically, then parallelly execute the graph nodes.

Comment (tonyyang-svail) [reply doc/support_multiple_sql_statements.md]: Sure. Added.

File: doc/support_multiple_sql_statements.md (hunk 1 above)
Comment (tonyyang-svail) [reply doc/support_multiple_sql_statements.md]: Sure. Updated.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**